kpops/components/base_components/models/to_section.py

```diff
@@ -15,15 +16,40 @@ class OutputTopicTypes(str, Enum):
 
 
 class TopicConfig(BaseModel):
-    """Configures a topic"""
+    """Configure an output topic
+
+    :param type: Topic type
+    :type type: InputTopicTypes
+    :param key_schema: Key schema class name
+    :type key_schema: str | None
+    :param partitions_count: Number of partitions into which the topic is divided
+    :type partitions_count: int | None
+    :param replication_factor: Replication topic of the topic
+    :type replication_factor: int | None
+    :param configs: Topic configs
+    :type configs: dict[str, str | int]
+    :param role: Custom identifier belonging to one or multiple topics, provide only if `type` is `extra`
+    :type role: str | None
+    """
 
-    type: OutputTopicTypes = Field(...)
-    key_schema: str | None = Field(default=None, alias="keySchema")
-    value_schema: str | None = Field(default=None, alias="valueSchema")
-    partitions_count: int | None = None
-    replication_factor: int | None = None
-    configs: dict[str, str] = {}
-    role: str | None = None
+    type: OutputTopicTypes = Field(..., description="Topic type")
+    key_schema: str | None = Field(
+        default=None, alias="keySchema", description="Key schema class name"
+    )
+    value_schema: str | None = Field(
+        default=None, alias="valueSchema", description="Value schema class name"
+    )
+    partitions_count: int | None = Field(
+        default=None, description="Number of partitions into which the topic is divided"
+    )
+    replication_factor: int | None = Field(
+        default=None, description="Replication topic of the topic"
+    )
+    configs: dict[str, str | int] = Field(default={}, description="Topic configs")
```

Comment:
we should validate that this change is working as expected. I had an issue recently in a similar Pydantic models, where the value type was defined as `str | bool` causing all bools to be cast to str :( I had to change the value type to `Any` as a workaround

Reply:
:/// How can I go about that?

Reply:
ideally we already have a snapshot test for a pipeline that uses topic configs. could you check if that's the case?
otherwise I can test it in a deployment project for you (and we can add a separate test for it)

Reply:
[tests/pipeline/resources/defaults.yaml](https://github.com/bakdata/kpops/blob/main/tests/pipeline/resources/defaults.yaml#L25) uses some